File: src/types/lyrics.ts
export interface LyricWord {
  time: number; // Time in milliseconds when this word starts
  text: string;
  endTime?: number; // Optional: when this word ends (for better accuracy)
}

export interface LyricLine {
  time: number; // Time in milliseconds when this line starts
  text: string;
  words?: LyricWord[]; // Optional: word-level timestamps if available
}

export interface LyricsData {
  lines: LyricLine[];
  synced: boolean;
  hasWordTimestamps?: boolean; // Whether word-level timestamps are available
}

File: tvOS/SpotifyLyricsTV/Models/Lyrics.swift
//
//  Lyrics.swift
//  SpotifyLyricsTV
//
//  Lyrics model matching the web app structure
//

import Foundation

struct LyricWord: Codable {
    let time: Int // Time in milliseconds
    let text: String
    let endTime: Int? // Optional end time
}

struct LyricLine: Codable {
    let time: Int // Time in milliseconds
    let text: String
    let words: [LyricWord]? // Optional word-level timestamps
}

struct LyricsData: Codable {
    let lines: [LyricLine]
    let synced: Bool
    let hasWordTimestamps: Bool?
}




